### DISCUSSION TITLE
A couple of ideas

### DISCUSSION THREAD

**Discussion by tutdrive77** [brianch/offline-chess-puzzles#16] [Ideas]
Hi,

First off, big thanks for all the hard work you've put into making this awesome software – it doesn't go unnoticed, it's really the one of a kind, to be able to play all the puzzles offline.

Had a couple of thoughts to make the puzzle app even more user-friendly:

1. Drag and Drop Functionality:
   - Instead of the current two-click method, it would be fantastic if we could just drag and drop pieces on the board. This feature would not only make the interface more intuitive but also look like the same option in Lichess.

2. Scan for New Pieces:
   - It would be cool if the program could scan the pieces folder for any new additions. For instance, I can save more pieces from Lucas Chess, and when the program starts these pieces appear in the drop-down menu.

3. Enhanced Link Functionality:
   - Adding an "Open" button alongside the puzzle link and game link (in the settings). This feature would allow users to open puzzles and games directly in their browser, providing a quicker and more convenient navigation option.

4. Book-Like Display Option:
   - Include an option for the viewer to emulate the appearance of a book. This could include a fixed board orientation (white at the bottom with the a1 square at the bottom left) and a clear indicator, like an arrow (black/white), on the right side, denoting the side to move. This feature would replicate the familiarity of working with puzzles in a chess book, as an extra option.

No pressure at all, just tossing in some thoughts for consideration. Keep up the fantastic work, and thanks again for making your hobby a chess haven for us!

Cheers,

**Comment by brianch** [comment]
Hello!

Sorry for the long delay, and thanks a lot for the nice words and ideas! :)

1) Drag and Drop: That would be the ideal indeed, when I started the project, the GUI library I use wasn't very suitable for this kind of thing and I wasn't familiar enough with it do it that way, but I would also like to have this, I just need to manage to do it lol

2) Scan for New Pieces: Oh, I hadn't noticed Lucas Chess had so many piece sets, that's a nice idea, I'll try to do it!

3) Enhanced Link Functionality: That's also a great suggestion, I will check if there's a nice way to do this in a platform independent way.

4) Book-Like Display Option: I don't think the view needs to be fixed, but this idea seems really cool! I'll take a look if I can make something like it.

I'm a bit busy with my studies and work, but in December I think I'll have much more time to work on this project again.

Thanks again for the great suggestions :)
